packages/xstate-svelte/src/fsm.ts

```diff
@@ -0,0 +1,53 @@
+import { readable } from 'svelte/store';
+import {
+  createMachine,
+  interpret,
+  EventObject,
+  StateMachine,
+  Typestate
+} from '@xstate/fsm';
+
+interface UseMachineOptions<
+  TContext extends object,
+  TEvent extends EventObject
+> {
+  /**
+   * If provided, will be merged with machine's `actions`.
+   */
+  actions: StateMachine.ActionMap<TContext, TEvent>;
+}
+
+export function useMachine<
+  TContext extends object,
+  TEvent extends EventObject,
+  TTypestate extends Typestate<TContext>
+>(
+  machine: StateMachine.Machine<TContext, TEvent, TTypestate>,
+  options: Partial<UseMachineOptions<TContext, TEvent>> = {}
+) {
+  const resolvedMachine = createMachine(machine.config, {
+    actions: {
```

Comment:
You can probably use `createMachine(machine.config).withOptions(...)` to handle the merging stuff for you.

Reply:
`withOptions` isn't available on `createMachine` in `@xstate/fsm`.